app/routes/__index/domains/new.tsx

```diff
@@ -1,11 +1,66 @@
-import { Heading } from '@chakra-ui/react';
+import { Container, Heading, Text } from '@chakra-ui/react';
+import { json, redirect } from '@remix-run/node';
+import { RecordType } from '@prisma/client';
+import { useActionData } from '@remix-run/react';
+import { z } from 'zod';
+import { parseFormSafe } from 'zodix';
+
+import DnsRecordForm from '~/components/dns-record/form';
+import { createRecord } from '~/models/record.server';
+import { requireUsername } from '~/session.server';
+
+import type { ActionArgs } from '@remix-run/node';
+import type { ZodError } from 'zod';
+
+function errorForField(error: ZodError, field: string) {
+  return error.issues.find((issue) => issue.path[0] === field)?.message;
+}
+
+export const action = async ({ request }: ActionArgs) => {
+  // Create a Zod schema for validation
+  // Optional is not needed as we get '' if nothing is entered
+  const DnsRecord = z.object({
+    name: z.string().min(1), // We do not want to consider '' a valid string
+    type: z.nativeEnum(RecordType),
+    value: z.string().min(1),
+    ports: z.string(),
```

Comment:
These three all need `.optional()` I think, see https://github.com/colinhacks/zod#optionals

Reply:
When I was testing this, the props would resolve to `''` ,aka an empty string, if the field was blank on the form. This is why I removed the `optional()` cause they would never not be present

Reply:
OK, sounds good.